packages/editor/src/components/inserter/menu.js

```diff
@@ -205,6 +205,17 @@ export class InserterMenu extends Component {
 			itemsPerCategory,
 			openPanels,
 		} );
+
+		const resultCount = Object.keys( itemsPerCategory ).reduce( ( accumulator, currentCategorySlug ) => {
```

Comment:
Coming to this way late, but in some recent refactoring I've come across remnants of this code. My first impression was: Why does this do a `reudce` operation on `itemsPerCategory` if `filteredItems` array is already available much more directly (i.e. `filteredItems.length`)? It then occurred to me that `itemsPerCategory` can yield slightly different results, notably in how it filters out reusable blocks. But then again, it doesn't seem to be that we'd _want_ reusable blocks to be excluded from the announced results.

In my testing, this is exactly what happens ("0 results found" when there's a reusable block search result present).

![image](https://user-images.githubusercontent.com/1779930/81603067-b2f78000-939b-11ea-9302-6243de861574.png)

Ultimately, then, it would seem that `filteredItems.length` would be both more direct and more accurate?

Reply:
> Ultimately, then, it would seem that `filteredItems.length` would be both more direct and more accurate?

Proposed for change at: #22279